src/resource.js

```diff
@@ -57,24 +58,51 @@ class Resource extends BaseResource {
     return null
   }
 
-  async count() {
-    return this.MongooseModel.countDocuments()
+  async count(filters) {
+    return this.MongooseModel.find(this.convertedFilters(filters)).countDocuments()
+  }
+
+  getDateFilter(filter) {
+    const { from, to } = filter
+    return {
+      ...from && { $gte: from },
+      ...to && { $lte: to },
+    }
+  }
+
+  getDefaultFilter(filter) {
+    return {
+      $regex: filter, $options: 'i',
+    }
+  }
+
+  convertedFilters(filters) {
+    if (!filters) return {}
+    return Object.keys(filters).reduce((obj, key) => {
+      const currentFilter = filters[key]
+      const isDateFilter = currentFilter.from || currentFilter.to
```

Comment:
maybe it would be clearer to remove getDefaultFilter and getDateFilter and put implementation to this method ? - to discuss